config.go

```diff
@@ -97,3 +113,49 @@ func choose(hosts Hosts) (*Host, int) {
 	}
 	return nil, -1
 }
+
+func findFirstNodeByQueue(currentHost *Host, hosts *Hosts, mutex *sync.RWMutex) (host *Host, error error) {
+	if len(*hosts) < 1 {
+		return nil, errors.New("list of hosts is empty")
+	}
+	hostMap := map[string]*Host{}
+	for v := range *hosts {
+		hostMap[fmt.Sprintf("%s%d%s", (*hosts)[v].Name, (*hosts)[v].Port, (*hosts)[v].Region)] = &(*hosts)[v]
+	}
+	var capacities []capacity
+	mutex.Lock()
+	defer mutex.Unlock()
+	var netClient = &http.Client{
+		Timeout: time.Second * 10,
+	}
+	for y := range *hosts {
+		rsp, err := netClient.Get((*hosts)[y].StatusEndPoint())
+		if err == nil {
+			responseMap := map[string]int{}
+			body, err := ioutil.ReadAll(rsp.Body)
+			if err == nil {
+				_ = json.Unmarshal(body, &responseMap)
+				capacities = append(capacities, capacity{&(*hosts)[y], responseMap["queued"], responseMap["pending"], responseMap["used"], responseMap["total"]})
```

Comment:
what if there no `queued`, `pending`, `used` or `total` field in json?
You should validate it first, or unmarshal to your `capacity` struct directly.